src/abstract-wallet.ts

```diff
@@ -650,6 +650,15 @@ export abstract class AbstractWallet {
         token: TokenLike,
         erc20ApproveThreshold: BigNumber = ERC20_APPROVE_TRESHOLD
     ): Promise<boolean> {
+
+        // validates token
+        try {
+            this.provider.tokenSet.resolveTokenSymbol(token)
+        } catch (error) {
+            throw error;
```

Comment:
What's the purpose of catching the error and throwing the same?

Reply:
Fair point. Though what I had in mind was to communicate at a glance that an error can be potentially thrown if validation fails, but you're right. It's redundant